frontend/src/app/organization/organization.component.ts

```diff
@@ -35,8 +35,11 @@ export class OrganizationComponent implements OnInit {
 
     this.activatedRoute.params.subscribe(
       params => {
-        this.organizationService.getOrganization(params.organizationId).subscribe();
-        this.organizationService.getOffersForOrganization(params.organizationId);
+        this.organizationService.getOrganization(params.organizationId)
+        .subscribe(
+          response => this.organizationService.getOffersForOrganization(params.organizationId),
```

Comment:
subscribe within subscribe often indicates something smells ;)

This change makes those two requests sequential instead of concurrent. I'm not sure what's the benefit.

Anyway, this part here (the way organizations are fetched and "injected" to the component) is really buggy (races...) and IMO needs to be changed to cancel the previous requests, but it's not the point of 404 page task.

Reply:
I know it's ugly, but the way it was before, it was still sending request to get organization's offers even if there wasn't organization with that id and I was getting errors in the console...